### PR TITLE
Recompute markers during cycles

### PR DESCRIPTION
**Pull request by jessecarterMOOSE**
Trying this again...
Adds a parameter to recompute markers during adaptivity cycles. Test case included. Ended up splitting computeIndicatorsAndMarkers() in two in order to just compute markers.

refs #6663

### CODE REVIEW

File: test/src/markers/CircleMarker.C [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+template<>
+InputParameters validParams<CircleMarker>()
+{
+  InputParameters params = validParams<Marker>();
+  params.addRequiredParam<Point>("point", "The center of the circle.");
+  params.addRequiredParam<Real>("radius", "Distance from the center of the circle to mark elements");
+
+  MooseEnum marker_states = Marker::markerStates();
+
+  params.addRequiredParam<MooseEnum>("inside", marker_states, "How to mark elements inside the box.");
```
Comment (friedmud) [inline test/src/markers/CircleMarker.C]: Small thing... but this still says `box` :-)

**Comment by friedmud** [comment]
I like it!

I like the CircleMarker.  It could probably go into MOOSE proper if you want... but it's fine to leave it in `test` for now.

Great job!  The tests aren't finished yet... so I won't merge it yet, but it can be merged in the morning for sure...

File: test/src/markers/CircleMarker.C (hunk 1 above)
Comment (jessecarterMOOSE) [inline test/src/markers/CircleMarker.C]: Whoops. I guess you can tell I stole this from BoxMarker :)

**Comment by permcody** [comment]
Thanks Jesse, This change causes some breakage in one of our internal applications but it's nothing we can't fixup on our own. I'll let you know if we need anything else from you but this looks good from our end.

**Comment by jessecarterMOOSE** [comment]
@friedmud Feel free to use CircleMarker however you wish. I could move it into the framework if you like, assuming all that's required is simply moving the files and unregistering/registering the object in the App file(s). Does it need its own test?

@permcody Glad to help. Let me know what else I can do to get this in there.

**Comment by dschwen** [comment]
Breaks Yak and rattlesnake.

File: framework/include/base/FEProblem.h [hunk 2]
```diff
@@ -727,7 +727,8 @@ class FEProblem :
                                 bool & changed_search_direction,
                                 bool & changed_new_soln);
 
-  virtual void computeIndicatorsAndMarkers();
```
Comment (dschwen) [inline framework/include/base/FEProblem.h]: Just don't remove this method! Let it in there and make it just call 

```
computeIndicators();
computeMarkers();
```

Comment (jessecarterMOOSE) [inline framework/include/base/FEProblem.h]: makes sense

**Comment by permcody** [comment]
Thanks again @jessecarterMOOSE. We'll get this merged

@idaholab/moose-developers does anyone have any strong feelings about whether we should deprecate `computeIndicatorsAndMarkers()`?

**Comment by jessecarterMOOSE** [comment]
@permcody Happy to help!

**Pull request merged**